src/objectid.ts

```diff
@@ -223,15 +219,15 @@ export class ObjectId {
       otherId.length === 12 &&
       isUint8Array(this.id)
     ) {
-      return otherId === Buffer.prototype.toString.call(this.id, 'latin1');
+      return ByteUtils.equals(this.id, ByteUtils.fromISO88591(otherId));
     }
 
     if (typeof otherId === 'string' && ObjectId.isValid(otherId) && otherId.length === 24) {
       return otherId.toLowerCase() === this.toHexString();
     }
 
     if (typeof otherId === 'string' && ObjectId.isValid(otherId) && otherId.length === 12) {
-      return Buffer.from(otherId).equals(this.id);
+      return ByteUtils.equals(ByteUtils.fromISO88591(otherId), this.id);
```

Comment:
I think this and the similar one above are currently unreachable because of https://github.com/mongodb/js-bson/pull/518/files#r1003675008?